independent-projects/resteasy-reactive/client/runtime/src/main/java/org/jboss/resteasy/reactive/client/impl/multipart/PausableHttpPostRequestEncoder.java

```diff
@@ -863,11 +859,14 @@ public HttpRequest finalizeRequest() throws ErrorDataEncoderException {
             if (request instanceof FullHttpRequest) {
                 FullHttpRequest fullRequest = (FullHttpRequest) request;
                 ByteBuf chunkContent = chunk.content();
-                if (fullRequest.content() != chunkContent) {
+                if (fullRequest.content() != chunkContent && chunkContent != null) {
                     fullRequest.content().clear().writeBytes(chunkContent);
                     chunkContent.release();
+                } else if (chunkContent == null) {
+                    isChunked = true;
+                    removeChunkedFromTransferEncoding(headers);
                 }
-                return fullRequest;
+                return new WrappedHttpRequest(fullRequest);
```

Comment:
Can you explain these changes please?

Reply:
Without these changes, `shouldSendFromMultiEmittingOutsideEventLoop` and `shouldUploadSlowlyProducedData` test cases will fail ([as you can see in the previous CI report](https://github.com/quarkusio/quarkus/pull/35982#issuecomment-1726349856)). For some reason to be determined, the fetched HTTP chunk is a null value and the request's content has to be marked as chunked. Otherwise, it throws NPE because the source buffer (i.e. the chunk) is a null value.

Reply:
Okay, I think we can live with it as long as CI passes and as long as https://github.com/quarkusio/quarkus/issues/35915 is fixed

Reply:
We could ask to the netty team what they think about these changes.

Reply:
Let's not burden them unless we really need to.